hoot-core/src/main/cpp/hoot/core/conflate/merging/LinearDiffMerger.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2021 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "LinearDiffMerger.h"
+
+// hoot
+#include <hoot/core/algorithms/linearreference/WaySublineRemover.h>
+#include <hoot/core/conflate/highway/HighwayMatch.h>
+#include <hoot/core/elements/RelationMemberUtils.h>
+#include <hoot/core/io/OsmMapWriterFactory.h>
+#include <hoot/core/ops/RemoveElementByEid.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/Log.h>
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(Merger, LinearDiffMerger)
+
+LinearDiffMerger::LinearDiffMerger() :
+LinearMergerAbstract(),
+_treatReviewsAsMatches(false)
+{
+}
+
+LinearDiffMerger::LinearDiffMerger(
+  const std::set<std::pair<ElementId, ElementId>>& pairs,
+  const std::shared_ptr<SublineStringMatcher>& sublineMatcher) :
+LinearMergerAbstract(pairs, sublineMatcher),
+_treatReviewsAsMatches(false)
+{
+}
+
+void LinearDiffMerger::apply(
+  const OsmMapPtr& map, std::vector<std::pair<ElementId, ElementId>>& replaced)
+{
+  LOG_TRACE("Applying LinearDiffMerger...");
+
+  _map = map;
+
+  // We're overriding some of LinearMergerAbstract's behavior here for diff purposes.
+
+  // This is the same as LinearMergerAbstract, except we don't require the associated element be
+  // part of the map in order to be merged (why?).
+  std::vector<std::pair<ElementId, ElementId>> pairs;
```

Comment:
My brain hasn't quite comprehended out why this part is necessary, but maybe it will at some point.